crates/compiler/gen_llvm/src/llvm/build.rs

```diff
@@ -2964,136 +2964,61 @@ fn list_literal<'a, 'ctx>(
     let list_length = elems.len();
     let list_length_intval = env.ptr_int().const_int(list_length as _, false);
 
-    // TODO re-enable, currently causes morphic segfaults because it tries to update
-    // constants in-place...
-    // if element_type.is_int_type() {
-    if false {
+    let is_all_constant = elems.iter().all(|e| e.is_literal());
+
+    if is_all_constant {
```

Comment:
Just a note, I think we will have to restrict this for now to `is_all_constant` and `does_not_contain_pointers`. I think the surgical linker does not currently handle constants that contain pointers. So no lists of lists, lists of strings, lists of recursive tags, etc.